### PR TITLE
Fixed buttons that are blocked too long 

### LINKED ISSUE #6061: Buttons are blocked a bit too long after e.g. going to the end page
#### ODK Collect version
the master version 031183e90dc4fa51ec389aa4987eeb048a6b470f
#### Android version
10, 14
#### Device used
Redmi 9T, Pixel 7a
#### Problem description
Buttons are blocked a bit too long after e.g. going to the end page in a form and tapping a button quickly. If a user taps the button at the end page quickly (after coming to the end page), it's necessary to tap the button twice.
The issue doesn't occur in the store version.

https://github.com/getodk/collect/assets/56479916/abc06599-f7f0-4ed0-bea4-46309a36398a

#### Steps to reproduce the problem
1. Open a form.
2. Go to the hierarchy view.
3. Tap "Go to end”.
4.  Tap "save as draft”/ "send” button quickly.
#### Expected behavior
The user shouldn't notice the time when the buttons are blocked.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #6061 

#### Why is this the best possible solution? Were any other approaches considered?
As discussed in the issue there are two approaches we can use depending on our needs:
1. We can stop using the `MultiClickSafeMaterialButton` if we have multiple buttons on one screen and we don't want to block them. Here a good example is the navigation buttons or the control buttons that we use to select/unselect/delete forms.
2. If we still want to block multi-clicking inside a particular screen but not if a user navigates from another screen then we need to make such buttons screen-aware: https://github.com/getodk/collect/pull/6098/commits/a4e84e4cf0953c76e2f05b30914e3a055e9bb273

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
This should fix the problem with the navigation buttons and the buttons that control lists of forms (select/unselect/delete). Nothing else should be affected. It's possible that other places in the app might need similar changes but we can identify them later and file separate issues.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] added or modified tests for any new or changed behavior
- [x] run `./gradlew connectedAndroidTest` (or `./gradlew testLab`) and confirmed all checks still pass
- [x] added a comment above any new strings describing it for translators
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: androidshared/src/main/java/org/odk/collect/androidshared/ui/multiclicksafe/MultiClickSafeMaterialButton.kt [hunk 1]
```diff
@@ -3,23 +3,32 @@ package org.odk.collect.androidshared.ui.multiclicksafe
...
-        attrs
-    )
-
-    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
-        context,
-        attrs,
-        defStyleAttr
-    )
+    init {
+        context.theme.obtainStyledAttributes(
```
Comment (seadowg) [inline androidshared/src/main/java/org/odk/collect/androidshared/ui/multiclicksafe/MultiClickSafeMaterialButton.kt]: Could we add a helper to `ContextUtils` for grabbing theme string theme attributes like this?

File: androidshared/src/main/res/values/screen_names.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (seadowg) [inline androidshared/src/main/res/values/screen_names.xml]: I feel like this should live in `collect_app` as it's about screens that are implemented there.

**Review by seadowg — CHANGES_REQUESTED** [review]
Really like the way this is done. I'll mark this as "needs testing" now as I've only requested a couple of tweaks that won't change behaviour.

File: androidshared/src/main/res/values/screen_names.xml (hunk 2 above)
Comment (grzesiek2010) [reply androidshared/src/main/res/values/screen_names.xml]: Done.

File: androidshared/src/main/java/org/odk/collect/androidshared/ui/multiclicksafe/MultiClickSafeMaterialButton.kt (hunk 1 above)
Comment (grzesiek2010) [reply androidshared/src/main/java/org/odk/collect/androidshared/ui/multiclicksafe/MultiClickSafeMaterialButton.kt]: Actually, kotlin offers an extension function called `withStyledAttributes` so we can use it https://github.com/getodk/collect/pull/6098/commits/ffa8c7990cbcc19879622d950c659937931726e4

**Review by seadowg — APPROVED** [review]

**Comment by srujner** [comment]
**Tested with Success!**

Verified on device with Android 12,13

Verified cases:
- Issue #6061 is no longer reproducing;
- Navigation buttons;
- Select/Deselect all buttons;
- Minimize the app, rotate the screen

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on device with Android 10

**Comment by WKobus** [comment]
**Tested with Success!**

Verified on device with Android 11

**Pull request merged**